modules/core/src/main/java/org/apache/ignite/internal/processors/query/GridQueryProcessor.java

```diff
@@ -2837,7 +2837,7 @@ private void checkxEnabled() throws IgniteException {
             throw new CacheException("Execution of local SqlFieldsQuery on client node disallowed.");
 
         if (experimentalQueryEngine != null && useExperimentalSqlEngine) {
-            if (!H2_REDIRECTION_RULES.matcher(qry.getSql()).find())
+            if (H2_REDIRECTION_RULES.matcher(qry.getSql()).find())
```

Comment:
we use such a check twice. Will be good to extract a method to check it?

Reply:
done